src/drive/ducks/files/Toolbar.jsx

```diff
@@ -24,27 +23,67 @@ import { uploadedFile, downloadFiles, trashFiles } from '../../actions'
 import {
   ShareButton,
   SharedWithMeButton,
+  SharedByMeButton,
   ShareModal,
   SharingDetailsModal
 } from 'sharing'
 import { leave, getSharingDetails } from 'cozy-client'
 
-const { BarRight } = cozy.bar
-
-const toggleShowShareModal = state => ({
-  ...state,
-  showShareModal: !state.showShareModal
-})
-
 const ALERT_LEVEL_INFO = 'info'
 const ALERT_LEVEL_ERROR = 'error'
 const ALERT_LEVEL_SUCCESS = 'success'
 
-class Toolbar extends Component {
+const ActionShare = props => {
+  const { highlighted, shared, onClick, t } = props
+  let label, Component
+
+  if (shared.withMe) {
+    label = 'Files.share.sharedWithMe'
+    Component = SharedWithMeButton
+  } else if (shared.byMe) {
+    label = 'Files.share.sharedByMe'
+    Component = SharedByMeButton
+  } else {
+    label = 'toolbar.share'
+    Component = ShareButton
+  }
+
+  return highlighted ? (
+    <Component label={t(label)} onClick={onClick} />
+  ) : (
+    <a className={styles['fil-action-share']} onClick={onClick}>
+      {t(label)}
+    </a>
+  )
+}
```

Comment:
We have a conditional assignement for `label` and `Component`. And a conditional `return`. We may find something clearer.